packages/cinterop/src/commonMain/kotlin/io/realm/internal/interop/ClassInfo.kt

```diff
@@ -16,16 +16,18 @@
 
 package io.realm.internal.interop
 
-// FIXME API-SCHEMA Platform independent property definition. Maybe rework into utility method
-//  called in Realm object's companion schema mechanism depending on how we relate this to the
-//  actual schema/runtime realm_property_info_t.
-@Suppress("LongParameterList")
-data class Property(
+data class ClassInfo(
     val name: String,
-    val publicName: String = "",
-    val type: PropertyType,
-    val collectionType: CollectionType = CollectionType.RLM_COLLECTION_TYPE_NONE,
-    val linkTarget: String = "",
-    val linkOriginPropertyName: String = "",
-    val flags: Set<PropertyFlag> = setOf(PropertyFlag.RLM_PROPERTY_NORMAL)
-)
+    val primaryKey: String?,
+    val numProperties: Long,
```

Comment:
I agree that count sounds better, but since this is a 1-1 representation of the similar Core struct `realm_class_info` I think it is ok to keep the names aligned.